contracts/rust/deployer/src/lib.rs

```diff
@@ -704,6 +826,46 @@ pub async fn is_contract(provider: impl Provider, address: Address) -> Result<bo
     Ok(true)
 }
 
+/// this depends on upgradeProxy.ts which has to be ran on a real network supported by the safeSDK
+pub async fn call_upgrade_proxy_script(
+    proxy_addr: String,
```

Comment:
Why not pass `Address` types? Can call `to_string()` on them to turn them into a string.